src/components/Editor/Utilities/ColorParser.vue

```diff
@@ -0,0 +1,172 @@
+        text += possible.charAt(Math.floor(Math.random() * possible.length));
+      }
+
+      return text;
+    },
+  },
+  computed: {
+    parsedText() {
+      const parts = this.text.split('&r');
+      for (var partIt = 0; partIt < parts.length; partIt++) {
+        var part = parts[partIt];
+
+        var hexBungeeMatches = part.match(/(&x&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9]))/gi);
+        if (hexBungeeMatches !== null) {
+          if (hexBungeeMatches.length > 0) part = part.replace(/(&x&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9])&([a-zA-Z0-9]))/gi, '<span style="color: #$2$3$4$5$6$7;">');
+          for (var i = 0; i < hexBungeeMatches.length; i++) part = part + '</span>';
+        }
+
+        var hexMatches = part.match(/(&#(([a-zA-Z0-9]){6}))/gi);
+        if (hexMatches !== null) {
+          if (hexMatches.length > 0) part = part.replace(/(&#(([a-zA-Z0-9]){6}))/gi, '<span style="color: #$2;">');
+          for (var i = 0; i < hexMatches.length; i++) part = part + '</span>';
+        }
+
+        var colorMatches = part.match(/(&([a-f0-9]))/gi);
+        if (colorMatches !== null) {
+          if (colorMatches.length > 0) part = part.replace(/(&([a-f0-9]))/gi, '<span class="mc-color-$2">');
+          for (var i = 0; i < colorMatches.length; i++) part = part + '</span>';
+        }
+
+        var boldMatches = part.match(/(&l)/gi);
+        if (boldMatches !== null) {
+          if (boldMatches.length > 0) part = part.replace(/(&l)/gi, '<span class="mc-bold">');
+          for (var i = 0; i < boldMatches.length; i++) part = part + '</span>';
+        }
+
+        var italicMatches = part.match(/(&o)/gi);
+        if (italicMatches !== null) {
+          if (italicMatches.length > 0) part = part.replace(/(&o)/gi, '<span class="mc-italic">');
+          for (var i = 0; i < italicMatches.length; i++) part = part + '</span>';
+        }
+
+        var underlineMatches = part.match(/(&n)/gi);
+        if (underlineMatches !== null) {
+          if (underlineMatches.length > 0) part = part.replace(/(&n)/gi, '<span class="mc-underline">');
+          for (var i = 0; i < underlineMatches.length; i++) part = part + '</span>';
+        }
+
+        var strikethroughMatches = part.match(/(&m)/gi);
+        if (strikethroughMatches !== null) {
+          if (strikethroughMatches.length > 0) part = part.replace(/(&m)/gi, '<span class="mc-strikethrough">');
+          for (var i = 0; i < strikethroughMatches.length; i++) part = part + '</span>';
+        }
+
+        var obfuscatedMatches = part.match(/(&k)/gi);
+        if (obfuscatedMatches !== null) {
+          if (obfuscatedMatches.length > 0) part = part.replace(/(&k)/gi, '<span class="mc-obfuscated">');
+          for (var i = 0; i < obfuscatedMatches.length; i++) part = part + '</span>';
+        }
+
+        parts[partIt] = part;
+      }
+      return parts.join('');
+    },
```

Comment:
Not going to go line by line on this one but a few suggestions for this computed function:

1. You can ditch the for loops and use `.forEach`
2. Use `const` and `let` instead of `var`
3. I think it would be good to do the `if` and `for` statements in curly braces `{ ... }` and use line breaks instead of doing the single line option, otherwise the lines can get too long and makes readability pretty low.
4. `part = part + '</span>'` can be simplified to `part += '</span>'`